pandas/core/arrays/categorical.py

```diff
@@ -376,9 +371,8 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
         null_mask = np.array(False)
 
         # sanitize input
-        if is_categorical_dtype(values):
-            if dtype.categories is None:
-                dtype = CategoricalDtype(values.categories, dtype.ordered)
+        if is_categorical_dtype(values) and dtype.categories is None:
```

Comment:
I think this should remain nested for perf reasons so the `elif` is not hit unnecessarily when the `is_categorical_dtype(values)` condition is met, but `dtype.categories is None` is not met.  I pulled your changes and did some quick timings with/without this being nested, and the difference does look to be significant.

Nested `if` statements (old version):
```python
In [2]: cat_old = pd.Categorical(list('abca')*10**4)

In [3]: cdt_new = pd.api.types.CategoricalDtype(list('bdac'), True)

In [4]: %timeit  pd.Categorical(cat_old, dtype=cdt_new)
799 µs ± 6.92 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Single flat `if` statement (new version):
```python
In [2]: cat_old = pd.Categorical(list('abca')*10**4)

In [3]: cdt_new = pd.api.types.CategoricalDtype(list('bdac'), True)

In [4]: %timeit  pd.Categorical(cat_old, dtype=cdt_new)
2.26 ms ± 12.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```